src/test/java/com/api/UserTests.java

```diff
@@ -106,52 +98,39 @@ public void getDeletedUserShouldReturn404Test() {
-        UserService userService = new UserService();
-        HttpResponse<String> firstResponse = userService.createUser(requestBody);
-        Assert.assertEquals(firstResponse.statusCode(), HttpStatus.CREATED.getCode(), "Expected status 201 when creating first user");
-        HttpResponse<String> secondResponse = userService.createUser(requestBody);
-        Assert.assertEquals(secondResponse.statusCode(), HttpStatus.UNPROCESSABLE_ENTITY.getCode(), "Expected status 422 for duplicate email");
-        Assert.assertTrue(secondResponse.body().contains("email"), "Expected 'email' field error message for duplicate email");
-    }
-
     @Test
     public void createUserWithEmptyBodyTest() {
         String emptyBody = "{}";
-        UserService userService = new UserService();
         HttpResponse<String> response = userService.createUser(emptyBody);
+        String responseBody = response.body();
         Assert.assertEquals(response.statusCode(), HttpStatus.UNPROCESSABLE_ENTITY.getCode(), "Expected status 422 when creating user with empty body");
-        Assert.assertTrue(response.body().contains("name"), "Expected validation error for missing 'name'");
-        Assert.assertTrue(response.body().contains("email"), "Expected validation error for missing 'email'");
-        Assert.assertTrue(response.body().contains("gender"), "Expected validation error for missing 'gender'");
-        Assert.assertTrue(response.body().contains("status"), "Expected validation error for missing 'status'");
+        assertUserFieldsPresent(responseBody);
     }
 
     @Test
     public void getNonExistingUserTest() {
-        int nonExistentUserId = 99999999;
-        UserService userService = new UserService();
-        HttpResponse<String> response = userService.getUserById(nonExistentUserId);
+        HttpResponse<String> response = userService.getUserById(NON_EXISTENT_USER_ID);
         Assert.assertEquals(response.statusCode(), HttpStatus.NOT_FOUND.getCode(), "Expected status 404 for non-existing user");
         Assert.assertTrue(response.body().contains("Resource not found"), "Expected 'Resource not found' message in response");
     }
+
+    private String getInvalidUserRequestBody() {
+        Map<String, String> invalidData = UserFactory.createInvalidUserData();
+        return RequestBodyBuilderUtil.buildUserRequestBody(invalidData);
+    }
+
+    private String getValidUserRequestBody() {
+        Map<String, String> userData = UserFactory.createDefaultUserData();
+        return RequestBodyBuilderUtil.buildUserRequestBody(userData);
+    }
+
+    private String getUpdatedUserRequestBody(Map<String, String> originalUserData) {
+        originalUserData.put("name", "Updated Name");
+        originalUserData.put("status", "inactive");
+        return RequestBodyBuilderUtil.buildUserRequestBody(originalUserData);
+    }
+
+    private Map<String, String> getValidUserData() {
+        return UserFactory.createDefaultUserData();
```

Comment:
Currently, helper methods like getValidUserRequestBody(), getInvalidUserRequestBody(), and getUpdatedUserRequestBody() are located directly inside the test class. While this works for smaller test suites, as the codebase grows, it's better to decouple data preparation logic from test logic for better maintainability and reuse.